source/elements/oneDAL/source/data_management/index.rst

```diff
@@ -5,12 +5,293 @@
+data. For |dal_short_name|, such set of operations, or **data management**, is
+distributed among different stages of a :ref:`data analytics pipeline
+<data_analytics_pipeline>`. From data management angle of view, this pipeline
+contains three main steps of data acquisition, preparation, and computation:
+
+1. Raw data acquisition.
+
+  - Transferring out-of-memory data into in-memory representation from sources
+    like databases, files, remote storages, etc.
+
+2. Data preparation.
+
+  - Performing data normalization, filtering.
+  - Recovery of missing values
+  - Compression and decompression of the data.
+  - Conversion of data to numeric representation.
+  - Supporting different in-memory :term:`data formats <Data format>`.
+  - Computation of statistics on numerical data like mean, variance,
+    covariance, etc.
+
+3. Algorithm computation.
+
+  - Streaming in-memory numerical data to the algorithm.
+
+In complex usage scenarios, data flow can go through these three stages back and
+forth. For example, when the full data are not available at the start of the
+computation, it can be done step-by-step with blocks of data. After computation
+on the current block the next one should be obtained and prepared.
+
+.. image:: _static/data_management_flow.png
+  :width: 800
+  :alt: Typical data management flow
+
+Key concepts
+============
+
+|dal_short_name| provides a set of concepts to operate on out-of-memory and
+in-memory data during different stages of the :ref:`data analytics pipeline
+<data_analytics_pipeline>`.
+
+.. _dataset:
+
+Dataset
+--------
+
+The most common concept over the data in |dal_short_name| is a :term:`dataset`.
+This is an in-memory or out-of-memory tabular view of data, where table rows
+represent :term:`observations <Observation>` and columns represent
+:term:`features <Feature>`.
+
+.. image:: _static/dataset.png
+  :width: 400
+  :alt: Dataset
+
+The dataset comes across all stages of the data analytics pipeline. For example:
+
+1. On the acquisition stage, it is downloaded into the
+   local memory
+
+2. On the preparation stage, it is converted to numerical
+   representation
+
+3. On the computation stage, it is used as one of the :ref:`inputs <Input>` or
+   :ref:`results <Result>` of an algorithm or even can be found in its
+   parameters
+
+.. _data-source:
+
+Data source
+-----------
+
+Data source is a concept of an out-of-memory storage of a :term:`dataset`. It is
+used on data acquisition and data preparation stages for:
+
+- Dataset extraction from databases, files, remote storages, etc.
+
+- Dataset loading into the local memory of the device. Data are not always fit
+  the local memory, especially when processing with accelerators. Data source
+  provides an ability to load data by batches and orchestrates its efficient
+  extraction directly into local memory of the device. Therefore data source
+  enables complex scenarios of data analytics, like :term:`online computations
```

Comment:
```suggestion
  the local memory, especially when processing with accelerators. A data source
  provides the ability to load data by batches and orchestrates its efficient
  extraction directly into the device's local memory. Therefore, a data source
  enables complex data analytics scenarios, such as :term:`online computations
```

`orchestrates its efficient extraction directly into` -> `efficiently extract it directly into`